GoogleCloudPlatform/terraform-google-analytics-lakehouse
# feat: add Cloudbuild config

This PR:

- adds a terraform cloudbuild config file to src/yaml (terraform.cloudbuild.yaml for init/plan/apply, and destroy.cloudbuild.yaml for init/destroy).
- creates terraform steps in setup.sh and destroy.sh to use cloudbuild instead of local terraform

To test:
- Open [this link](https://ssh.cloud.google.com/cloudshell/editor?cloudshell_git_repo=https://github.com/hyunuk/terraform-google-analytics-lakehouse/) to clone the forked branch in your Cloud Shell and check `Trust repo`
  - You must be able to log in with the appropriate account. If not, execute `gcloud auth login`.
- run `git checkout cloudbuild`
- run `./setup.sh project_id="PROJECT_NAME" region="us-central1" enable_apis=true force_destroy=true`. The script will execute in following instructions.
  - Check if terraform.tfvars exists and read project name and region if available.
    - If it exists, get project name and region from the file and store them into variables.
    - If not, create terraform.tfvars file based on parameters.
  - Check if backend.tf exists and create a new one if it isn't. backend.tf file is necessary to set a bucket to store terraform state file.
  - Enable services that are needed to manage resources.
  - Grant IAM permissions to the cloudbuild service account
  - Setup Terraform state bucket
  - Run Cloud build
- after applying, run `./destroy` to destroy terraform deployment and revoke permissions.
